File: Tech/ResponseBodyAdviceExample/src/main/java/com/kgr/responsebodyadviceexample/controller/MainController.java
package com.kgr.responsebodyadviceexample.controller;

import com.kgr.responsebodyadviceexample.core.model.ResponseWrapper;
import com.kgr.responsebodyadviceexample.service.MainService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class MainController {

    private final MainService mainService;

    @GetMapping("/test")
    public ResponseWrapper insertData(){
        return mainService.insertData();
    }
}

File: Tech/SpringKotlin/src/main/kotlin/com/kotlin/test/controller/MainController.kt
package com.kotlin.test.controller

import com.kotlin.test.model.StudentModel
import com.kotlin.test.service.TestService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class MainController(
    private val testService: TestService
) {
    @GetMapping("/")
    fun test(): String {
        return "test"
    }

    @PostMapping("/insertUserInfo")
    fun insertUserInfo(@RequestBody studentModel: StudentModel): StudentModel{
        return testService.insertUserInfo(studentModel)
    }
}